tests/Fixer/Phpdoc/PhpdocTypesOrderFixerTest.php

```diff
@@ -391,6 +391,18 @@ public function provideFixWithAlphaAlgorithmCases(): array
             [
                 '<?php /** @return array<int, callable(array<string, null|string> , DateTime): bool> */',
             ],
+            [
+                '<?php /** @return A&B&C */',
+                '<?php /** @return A&C&B */',
+            ],
+            [
+                '<?php /** @return array<A&B&C>|bool|string */',
+                '<?php /** @return bool|array<A&B&C>|string */',
```

Comment:
I would suggest you to add a case where parentheses are involved.. It's a much more common case when intersection and union are mixed, since it's the only way to mix them up.

Reply:
Do you mean something like:
```php

            [
                '<?php /** @return (A|B)&C */',
                '<?php /** @return (A|C)&B */',
            ],
            [
                '<?php /** @return (A&B)|C */',
                '<?php /** @return (A&C)|B */',
            ],
```
This both cases fail right now as TypeEpression considers parentheses as part of type.

Reply:
Considering part of the type is correct: changing the position like that changes the meaning/shape of the type. I was thinking something like:
```
            [
                '<?php /** @return (A|B)&C */',
                '<?php /** @return (B|A)&C */',
            ],
            [
                '<?php /** @return (A&B)|C */',
                '<?php /** @return (C&A)|B */',
            ],
```

Reply:
I think
```
            [
                '<?php /** @return (A&C)|B */',
                '<?php /** @return (C&A)|B */',
            ],
```

is this something that was changed in this PR or was not supported and not added in the PR? Do we want/need it in this PR?

Reply:
It was not supported and is not supported in this PR. Let's make this PR as bug fix and add support for these in separate PR.

Reply:
thanks, sounds good to me :+1: